pager/src/main/java/com/google/accompanist/pager/PagerState.kt

```diff
@@ -285,6 +327,38 @@ class PagerState(
         snapToNearestPage()
     }
 
+    private suspend fun animateToPageSkip(
+        page: Int,
+        pageOffset: Float,
+        animationSpec: AnimationSpec<Float>,
+        initialVelocity: Float,
+    ) {
+        // Set our target page
+        _animationTargetPage = page
+
+        val initialIndex = currentLayoutPage
+        val pages: IntArray = when {
+            page > initialIndex -> intArrayOf(initialIndex, initialIndex + 1, page - 1, page)
+            else -> intArrayOf(initialIndex, initialIndex - 1, page + 1, page)
+        }
+
+        animate(
+            initialValue = currentPageOffset,
+            targetValue = pages.size + pageOffset - 1,
+            initialVelocity = initialVelocity,
+            animationSpec = animationSpec
+        ) { value, _ ->
+            val flooredIndex = floor(value).toInt()
+            layoutPages.forEachIndexed { index, layoutInfo ->
```

Comment:
Add a comment on what this is doing. I *think* it's going through and updating `layoutPages` to only the pages you want to scroll through but took some staring at.

Reply:
Added